src/app/components/core/configuration-manager/show-configuration/show-configuration.component.ts

```diff
@@ -0,0 +1,135 @@
+import { Observable, of } from 'rxjs';
+import { filter, map, pairwise, startWith } from 'rxjs/operators';
+import { RolesService, ConfigurationControlService, ConfigurationBase } from '../../../../services';
+
+@Component({
+  selector: 'app-show-configuration',
+  templateUrl: './show-configuration.component.html',
+  styleUrls: ['./show-configuration.component.css']
+})
+export class ShowConfigurationComponent implements OnInit {
+  @Input() fullConfiguration: any;
+  @Input() groupConfiguration: ConfigurationBase<string>[] | null = [];
+  @Input() group: string = '';
+  @Input() selectedGroup = '';
+
+  @Output() event = new EventEmitter<any>();
+  @Output() formStatusEvent = new EventEmitter<boolean>();
+  configurations$: Observable<ConfigurationBase<any>[]>;
+  form: FormGroup;
+
+  @ViewChildren('scriptCode') codeMirrorCmpt: QueryList<CodemirrorComponent>;
+  @ViewChildren('jsonEditor') josnElements: QueryList<CodemirrorComponent>;
+  @ViewChildren('codeEditor') codeElements: QueryList<CodemirrorComponent>;
+
+  constructor(private fb: FormBuilder,
+    public rolesService: RolesService,
+    public changeDetectorRef: ChangeDetectorRef,
+    private configControlService: ConfigurationControlService) {
+    this.form = this.fb.group({
+    });
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (!changes?.selectedGroup?.firstChange) {
+      if (changes?.selectedGroup?.currentValue == this.group) {
+        this.configControlService.checkConfigItemOnGroupChange(this.form, this.fullConfiguration);
+      }
+
+      if (this.josnElements) {
+        this.josnElements.forEach((jsonComp: CodemirrorComponent) => {
+          jsonComp.codeMirror.refresh();
+        });
+      }
+
+      if (this.codeElements) {
+        this.codeElements.forEach((codeElmt: CodemirrorComponent) => {
+          codeElmt.codeMirror.refresh();
+        });
+      }
+      // refresh codemirror editor to reflect changed values
+      if (this.codeMirrorCmpt) {
+        this.codeMirrorCmpt.forEach((comp: CodemirrorComponent) => {
+          comp.codeMirror?.refresh();
+        })
+      }
+    }
+  }
+
+  ngOnInit(): void {
+    this.groupConfiguration = this.configControlService.createConfigurationBase(this.groupConfiguration);
+    this.configurations$ = of(this.groupConfiguration);
+    this.form = this.configControlService.toFormGroup(this.fullConfiguration, this.groupConfiguration as ConfigurationBase<string>[]);
+    this.form.valueChanges.pipe(
+      startWith(this.form.value),
+      pairwise(),
+      map(([oldState, newState]) => {
+        let changes = {};
+        for (const key in newState) {
+          if (oldState[key] !== newState[key]) {
+            changes[key] = newState[key];
+          }
+        }
+        return changes;
+      }),
+      filter(changes => Object.keys(changes).length !== 0)
+    ).subscribe(
+      data => {
+        const [key, value] = Object.entries(data)[0];
```

Comment:
data length guaranteed?

Reply:
yes, only the changed value returned in this value change subscriber.